src/main/java/bio/terra/dao/FileDao.java

```diff
@@ -0,0 +1,324 @@
+package bio.terra.dao;
+
+import bio.terra.dao.exception.CorruptMetadataException;
+import bio.terra.dao.exception.FileSystemObjectAlreadyExistsException;
+import bio.terra.dao.exception.FileSystemObjectDependencyException;
+import bio.terra.dao.exception.FileSystemObjectNotFoundException;
+import bio.terra.dao.exception.InvalidFileSystemObjectTypeException;
+import bio.terra.metadata.FSObject;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Repository
+public class FileDao {
```

Comment:
i think this should be in a separate package (maybe bio.terra.fsdao). we can change the package name for bio.terra.dao to something else to make is more specific. And I think this class should be called FSObjectDao (since it is handling files and directories)

Reply:
As per our discussion this morning, I moved the FileDao into a `filesystem` package.